packages/fields/src/actions/set-as-homepage.tsx

```diff
@@ -0,0 +1,192 @@
+					page_on_front: item.id,
+					show_on_front: 'page',
+				} );
+
+				createSuccessNotice( __( 'Homepage updated' ), {
+					type: 'snackbar',
+				} );
+
+				onActionPerformed?.( items );
+			} catch ( error ) {
+				const typedError = error as CoreDataError;
+				const errorMessage =
+					typedError.message && typedError.code !== 'unknown_error'
+						? typedError.message
+						: __( 'An error occurred while setting the homepage' );
+				createErrorNotice( errorMessage, { type: 'snackbar' } );
+			}
+		}
+
+		const renderModalBody = () => {
+			if ( 'posts' === showOnFront ) {
+				return (
+					<>
+						<Text>
+							{ sprintf(
+								// translators: %s: title of the page to be set as the homepage.
+								__(
+									'Set "%s" as the site homepage? This will replace the current homepage which is set to display latest posts.'
+								),
+								pageTitle
+							) }
+						</Text>
+					</>
+				);
+			}
+
+			const modalTranslatedString =
+				// translators: %1$s: title of page to be set as the home page. %2$s: title of the current home page.
+				__(
+					'Set "%1$s" as the site homepage? This will replace the current homepage: "%2$s"'
+				);
+
+			return (
+				<Text>
+					{ sprintf(
+						modalTranslatedString,
+						pageTitle,
+						currentHomePageTitle
+					) }{ ' ' }
+				</Text>
+			);
+		};
+
+		return (
+			<form onSubmit={ onSetPageAsHomepage }>
+				<VStack spacing="5">
+					{ renderModalBody() }
+					<HStack justify="right">
+						<Button
+							__next40pxDefaultSize
+							variant="tertiary"
+							onClick={ () => {
+								closeModal?.();
+							} }
+						>
+							{ __( 'Cancel' ) }
+						</Button>
+						<Button
+							__next40pxDefaultSize
+							variant="primary"
+							type="submit"
+						>
+							{
+								// translators: Button to confirm setting the specified page as the homepage.
+								__( 'Set homepage' )
+							}
+						</Button>
+					</HStack>
+				</VStack>
+			</form>
+		);
+	};
+
+const setAsHomepage: Action< PostWithPermissionsAndContext > = {
+	id: 'set-as-homepage',
+	label: __( 'Set as homepage' ),
+	isEligible( post ) {
+		const { pageOnFront, pageForPosts } =
+			post.additionalContext?.siteSettings;
+		const { hasFrontPageTemplate } = post.additionalContext?.themeInfo;
+
+		if ( post.status === 'trash' ) {
+			return false;
+		}
+
+		if ( post.type !== PAGE_POST_TYPE ) {
+			return false;
+		}
+
+		// A front-page template overrides homepage settings, so don't show the action if it's present.
+		if ( hasFrontPageTemplate ) {
+			return false;
+		}
+
+		// Don't show the action if the page is already set as the homepage.
+		if ( pageOnFront === post.id ) {
+			return false;
+		}
+
+		// Don't show the action if the page is already set as the page for posts.
+		if ( pageForPosts === post.id ) {
+			return false;
+		}
+
+		return true;
+	},
+	RenderModal: SetAsHomepageModal,
+};
+
+export default setAsHomepage;
```

Comment:
Small thing: if you a JSDoc to this export, it'll fill the README above that it now says "Undocumented declaration".